### ISSUE TITLE
`borg with-lock` raises `FileNotFoundError`

### ISSUE THREAD

**Issue by kmille** [borgbackup/borg#8022]
Hey,
I just want to find out what `with-lock` does. I tried some things and I ended up with an exception :)

```bash
kmille@linbox:~ borg --verbose with-lock /home/kmille/tmp/borgrepo list
Local Exception
Traceback (most recent call last):
  File "/usr/lib/python3.11/site-packages/borg/archiver.py", line 5343, in main
    exit_code = archiver.run(args)
                ^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/borg/archiver.py", line 5263, in run
    return set_ec(func(args))
                  ^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/borg/archiver.py", line 189, in wrapper
    return method(self, args, repository=repository, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/borg/archiver.py", line 1853, in do_with_lock
    return subprocess.call([args.command] + args.args, env=env)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/subprocess.py", line 389, in call
    with Popen(*popenargs, **kwargs) as p:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/subprocess.py", line 1026, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/usr/lib/python3.11/subprocess.py", line 1950, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'list'

Platform: Linux linbox 6.6.8-hardened1-1-hardened #1 SMP PREEMPT_DYNAMIC Thu, 21 Dec 2023 23:45:24 +0000 x86_64
Linux: Unknown Linux  
Borg: 1.2.7  Python: CPython 3.11.6 msgpack: 1.0.5 fuse: llfuse 1.5.0 [pyfuse3,llfuse]
PID: 186644  CWD: /home/kmille
sys.argv: ['/usr/bin/borg', '--verbose', 'with-lock', '/home/kmille/tmp/borgrepo', 'list']
SSH_ORIGINAL_COMMAND: None
```

I don't understand the [docs](https://borgbackup.readthedocs.io/en/stable/usage/lock.html). After reading, I ask myself:
- What's the purpose of a lock if I can run other borg commands in the meantime
- "It will first try to acquire the lock": Does this just mean: wait until it's free?
- 

Maybe it makes sense to update the docs :)

**Comment by ThomasWaldmann** [comment]
The purpose of the lock is to lock the repo against other usage, e.g. if you want to some command on it and make sure it is in a steady state and does not change.

E.g.:
```
borg with-lock /mnt/borgrepo rsync -av /mnt/borgrepo /somewhere/else
```

**Comment by ThomasWaldmann** [comment]
Can you make a PR and add the example to the docs?

**Comment by ThomasWaldmann** [comment]
Another useful change would be to catch the `FileNotFoundError` here and emit an error message without traceback.

**Comment by kmille** [comment]
Thanks for the explanation! I will try tomorrow!

**Comment by kmille** [comment]
I fail to reproduce the issue with the master branch. I'm not able to use borg at all...?
I can not use my current repo.
```bash
(borg-env) kmille@linbox:borg borg-env/bin/borg -r /home/kmille/tmp/borgrepo list
usage: borg [-V] [-h] [--critical] [--error] [--warning] [--info] [--debug] [--debug-topic TOPIC] [-p] [--iec] [--log-json] [--lock-wait SECONDS] [--bypass-lock] [--show-version] [--show-rc] [--umask M] [--remote-path PATH]
            [--upload-ratelimit RATE] [--upload-buffer UPLOAD_BUFFER] [--debug-profile FILE] [--rsh RSH] [--socket [PATH]] [-r REPO]
            <command> ...
borg: error: argument -r/--repo: invalid validator value: '/home/kmille/tmp/borgrepo'
```
And can't create a new one ...?
```bash
(borg-env) kmille@linbox:borg borg-env/bin/borg -r ~/tmp/borgrepo2 rcreate
usage: borg [-V] [-h] [--critical] [--error] [--warning] [--info] [--debug] [--debug-topic TOPIC] [-p] [--iec] [--log-json] [--lock-wait SECONDS] [--bypass-lock] [--show-version] [--show-rc] [--umask M] [--remote-path PATH]
            [--upload-ratelimit RATE] [--upload-buffer UPLOAD_BUFFER] [--debug-profile FILE] [--rsh RSH] [--socket [PATH]] [-r REPO]
            <command> ...
borg: error: argument -r/--repo: invalid validator value: '/home/kmille/tmp/borgrepo2'
```
I'm following the quickstart ...

**Comment by ThomasWaldmann** [comment]
master branch will be borg2 at some time and is very different.

1.2.x code is in 1.2-maint branch, upcoming 1.4 (will supersede 1.2 in the near future) code is in 1.4-maint branch.

**Comment by kmille** [comment]
Okay? 
1) How can I run borg without running into the `invalid validator value` issue?
2) Which branch should I use for my PR? That's still not clear to me.
3) I need to close this PR and open a new for a different branch, do I?

**Comment by ThomasWaldmann** [comment]
1. use 1.2-maint branch (that is closest to what you practically use)
2. 1.2-maint
3. yes

**Comment by kmille** [comment]
Sorry for the confusion

**Comment by kmille** [comment]
Fixed in https://github.com/borgbackup/borg/commit/3551f03af45145146e6d56c70936f546bc9f2e6b

**Comment by ThomasWaldmann** [comment]
Fix for 1.4-maint coming in #8030.
